client/state/posts/actions.js

```diff
@@ -252,7 +258,37 @@ export function savePost( siteId, postId = null, post ) {
  * @return {Function}        Action thunk
  */
 export function trashPost( siteId, postId ) {
-	return savePost( siteId, postId, { status: 'trash' } );
+	return dispatch => {
+		const post = { status: 'trash' };
+
+		dispatch( {
+			type: POST_SAVE,
+			siteId,
+			postId,
+			post,
+		} );
+
+		const trashResult = wpcom
+			.site( siteId )
+			.post( postId )
+			.delete();
+
+		trashResult
+			.then( savedPost => {
+				dispatch( savePostSuccess( siteId, postId, savedPost, post ) );
```

Comment:
Why do we need to call `savePostSuccess` here?  It looks like the only thing it does is update the post ID for a newly saved draft from `''` (used by the editor) to the newly created ID, but trashing a post that doesn't actually exist yet shouldn't be possible.

Reply:
I moved this part of the changes to a separate PR, because it's fairly independent: #23359. Its description answers most questions.

Here, I'm refactoring the `trashPost` action to use the `wpcom.site().post().delete()` method, but keep its behavior the same as if it was `savePost( { status: 'trash' } )` -- the actions dispatched are the same etc.

This is to avoid too many changes across the app. We don't have any `POST_TRASH_SUCCESS` Redux actions, and there are things that depend on dispatching `POST_SAVE_SUCCESS` after trashing. For example, see [this code that displays a global notice](https://github.com/Automattic/wp-calypso/blob/master/client/state/notices/middleware.js#L131-L146).

Reply:
SGTM - I missed some other effects of the `savePostSuccess` action, like sending notices.  I'll look at that PR separately.